### DISCUSSION TITLE
How to add only styles to a Mantine Component and extends all props?

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#5252] [Q&A] [answered]
Take `Button` for example:
I'd like to define a button:
```
interface AnchorButtonProps extends ButtonProps

const MyButton = (props: MyButtonProps) =>
<Button
  variant="transparent"
  size="sm"
  className={classes.navButton}
  classNames={{ label: classes.label }}
  {...props}
/>
```
But this button won't work with some of the props of `Button`, like `component` and `onClick`,
it complains `Property 'component' does not exist on type 'IntrinsicAttributes & MyButtonProps'`

Is there a `ButtonProps` to `extends` so that all props work?

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/guides/polymorphic/#create-your-own-polymorphic-components
